lib/contentTypeParser.js

```diff
@@ -216,13 +226,36 @@ function buildContentTypeParser (c) {
   return contentTypeParser
 }
 
-function Parser (asString, asBuffer, bodyLimit, fn) {
-  this.asString = asString
-  this.asBuffer = asBuffer
-  this.bodyLimit = bodyLimit
-  this.fn = fn
+function addContentTypeParser (contentType, opts, parser) {
+  if (this[kState].started) {
+    throw new Error('Cannot call "addContentTypeParser" when fastify instance is already started!')
+  }
+
+  if (typeof opts === 'function') {
+    parser = opts
+    opts = {}
+  }
+
+  if (!opts) opts = {}
+  if (!opts.bodyLimit) opts.bodyLimit = this[kBodyLimit]
+
+  if (Array.isArray(contentType)) {
+    contentType.forEach((type) => this[kContentTypeParser].add(type, opts, parser))
+  } else {
+    this[kContentTypeParser].add(contentType, opts, parser)
+  }
+
+  return this
+}
+
+function hasContentTypeParser (contentType) {
+  return this[kContentTypeParser].hasParser(contentType)
 }
 
 module.exports = ContentTypeParser
-module.exports.buildContentTypeParser = buildContentTypeParser
+module.exports.helpers = {
```

Comment:
Are these helper functions exposed to the public API or just meant to be used internally? If they are going to be used externally I'd request we add some types; if not, no types are needed

Reply:
The public API is the big `fastify` object you can see in the main faile :)